week2/project/ecommerce/src/components/CategoryClicked.jsx

```diff
@@ -0,0 +1,59 @@
+import { useState, useEffect } from "react";
+import CategoryList from "./CategoryList.jsx";
+import Products from "./Products.jsx";
+
+export default function CategoryClicked() {
+  const [selectedCategory, setSelectedCategory] = useState(null);
+  const [allCategories, setCategories] = useState([]);
+  const [allProducts, setProducts] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        setLoading(true)
+        const responseCategories = await fetch(
+          "https://fakestoreapi.com/products/categories"
+        );
+        setCategories(await responseCategories.json());
+
+        if (!selectedCategory) {
+          const responseProducts = await fetch(
+            "https://fakestoreapi.com/products"
+          );
+          setProducts(await responseProducts.json());
+        } else {
+          const responseProducts = await fetch(
+            `https://fakestoreapi.com/products/category/${selectedCategory}`
+          );
+          setProducts(await responseProducts.json());
+        }
+      } catch (e) {
+        setError(e);
+      }finally{
+        setLoading(false)
+      }
+    }
+    fetchData();
+  }, [selectedCategory]);
+
+  const handleCategoryClick = (category) => {
+    if (selectedCategory === category) {
+      setSelectedCategory(null);
+    } else {
+      setSelectedCategory(category);
+    }
+  };
+
+  return (
+    <div>
+      <CategoryList
+        selectedCategory={selectedCategory}
+        list={allCategories}
+        onCategoryClick={handleCategoryClick}
+      />
+      {loading ? <p>Loading...</p> : error ? <h1>Something went wrong!!</h1>: <Products list={allProducts} />}
+    </div>
+  );
+}
```

Comment:
Nice job adding loading and error states! 🙌 

To make this a bit more readable you can also render the loading and error UI above:

```suggestion
  if (loading) {
    return <p>Loading...</p>
  }
  
  if (error) {
    return <h1>Something went wrong</h1>
  }

  return (
    <div>
      <CategoryList
        selectedCategory={selectedCategory}
        list={allCategories}
        onCategoryClick={handleCategoryClick}
      />
     <Products list={allProducts} />
    </div>
  );
}
```